src/main/java/org/opensearch/timeseries/transport/JobRequest.java

```diff
@@ -15,13 +15,15 @@
 
 import org.opensearch.action.ActionRequest;
 import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.action.DocRequest;
 import org.opensearch.core.common.io.stream.StreamInput;
 import org.opensearch.core.common.io.stream.StreamOutput;
 import org.opensearch.timeseries.model.DateRange;
 
-public class JobRequest extends ActionRequest {
+public class JobRequest extends ActionRequest implements DocRequest {
```

Comment:
this is how resource access requests are filtered for evaluation. Check out implementation here: https://github.com/opensearch-project/security/pull/5408/files#diff-5892c7b9dfd08da869c293e469b09654309be9994205ca939f0ce9e336d529de